app/controllers/two_factor_authentication_controller.rb

```diff
@@ -39,51 +45,29 @@ def verify_auth_token_and_redirect(token)
 
         flash[:notice] = "Successfully logged in!"
 
-        respond_to do |format|
-          format.html { redirect_to login_path_for(@user) }
-          format.json { render json: { redirect_location: login_path_for(@user) } }
-        end
+        redirect_to login_path_for(@user), status: :see_other
       else
-        respond_to do |format|
-          format.html do
-            flash[:alert] = "Invalid token, please try again."
-            redirect_to two_factor_authentication_path
-          end
-          format.json { render json: { error_message: "Invalid token, please try again." }, status: :unprocessable_entity }
-        end
+        redirect_to two_factor_authentication_path, warning: "Invalid token, please try again.", status: :see_other
```

Comment:
Let's keep a 302 response for redirections upon failure
```suggestion
        redirect_to two_factor_authentication_path, warning: "Invalid token, please try again."